src/core_tests.rs

```diff
@@ -178,6 +187,27 @@ impl TestNode {
     }
 }
 
+// Prints all nodes in a table, organised by group.
+// Does not check the routing tables (use verify_invariant_for_all_nodes() for that).
+fn print_nodes(nodes: &[TestNode], clients: &[TestClient]) {
```

Comment:
I prefer a paradigm where we print things like this inside `drop()` functions if the current thread is panicing.  E.g. I was planning to submit this in a PR for RT:
```rust
impl<T: Binary + Clone + Copy + Debug + Default + Hash + Xorable> Drop for RoutingTable<T> {
    fn drop(&mut self) {
        if thread::panicking() {
            trace!("{:?}", self);
        }
    }
}
```
If that's not preferred either, then I agree that we shouldn't merge unused code like this.